### PR TITLE
Make log and workspace dirs use the same timestamp

### PR DESCRIPTION
**Pull request by bluebytes60**

### CODE REVIEW

File: dir_test.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	return true, nil
+}
+
+
+func builtForHook(tempDir, owner, repo string, exitCode int) error {
+	return onHook("not-used", &effectiveConfig{resultRoot: tempDir, workspaceRoot:tempDir}, hookEvent{Owner: owner, Repo: repo}, StubBuild)
+}
+
+func StubBuild(configRoot string, resultPath string, config *effectiveConfig, hook hookEvent, basename string) (*executeResult, string, error) {
+	fmt.Print("result path:"+resultPath+"\n")
```
Comment (kklipsch) [inline dir_test.go]: dont want printlns in tests.

File: dir_test.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	return grimBuild(ws, resultPath, basename)
+}
+
+type testWorkSpaceBuilder struct {
+	workspaceBuilder
+}
+
+func (tb *testWorkSpaceBuilder) PrepareWorkspace(basename string) (string, error) {
+	workSpacePath, err := createWorkspaceDirectory(tb.workspaceRoot, tb.owner, tb.repo, basename)
+	fmt.Print("workspace:"+workSpacePath)
```
Comment (kklipsch) [inline dir_test.go]: ditto

File: execute_result.go [hunk 3]
```diff
@@ -22,6 +22,7 @@ type executeResult struct {
 	Output     string `json:"-"`
 }
 
+//append the results to "result.json" file
```
Comment (kklipsch) [inline execute_result.go]: not a terribly useful comment.

**Comment by kklipsch** [comment]
I think we need to squash some commits before merging this in.

**Comment by bluebytes60** [comment]
let me try to squash some of these commits

File: build_test.go [hunk 4]
```diff
@@ -1,155 +1,155 @@
...
+func (tb *testBuilder) RunBuildScript(workspacePath, buildScript string, outputChan chan string) (*executeResult, error) {
+	return tb.buildResult, tb.buildErr
+}
+
+func TestOnBuildStatusFileError(t *testing.T) {
+	resultPath, _ := ioutil.TempDir("", "build-status-file-error")
+	defer os.RemoveAll(resultPath)
+
+	tb := &testBuilder{workspaceErr: errors.New(""), workspaceResult: "@$@$"}
+	grimBuild(tb, resultPath,"")
```
Comment (bhand-mm) [inline build_test.go]: See if you can configure your editor to run 'goimports' and 'golint' on save.  It will keep all our code formatted the same.

**Comment by kklipsch** [comment]
Closing request as there is another one.

**Pull request closed without merging**